File: ExcludedSamples/NetFramework/Ribbon/VB/ControlHost/DemoForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.C1Ribbon
Imports ControlHostLib

Public Class DemoForm

    Private _internalFontChange As Boolean

    Public Sub New()

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        Dim f As Font = label1.Font
        _internalFontChange = True
        fontEditor.SelectedFont = f
        numericUpDownHost1.Value = Convert.ToDecimal(f.SizeInPoints)
        _internalFontChange = False

        backColorEditor.SelectedColor = label1.BackColor
        foreColorEditor.SelectedColor = label1.ForeColor

        axWebBrowserHost1.Navigate(Application.StartupPath + "\..\..\Html\A plea for C++.htm")

    End Sub

    Private Sub exitButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles exitButton.Click, RibbonApplicationMenu1.DoubleClick
        Close()
    End Sub

    Private Sub pictureBoxHost1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles pictureBoxHost1.Click
        MessageBox.Show("You clicked the Earth.", "Congratulations", MessageBoxButtons.OK, MessageBoxIcon.Information)
    End Sub

    Private Sub numericUpDownHost1_ValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles numericUpDownHost1.ValueChanged
        If Not _internalFontChange Then
            Dim f As Font = label1.Font
            f = New Font(f.Name, Convert.ToSingle(numericUpDownHost1.Value), f.Style, f.Unit)
            _internalFontChange = True
            label1.Font = f
            fontEditor.SelectedFont = f
            _internalFontChange = False
        End If
        statusLabel.Text = String.Format("Number is {0}", numericUpDownHost1.Value)
    End Sub

    Private Sub checkedListBoxHost1_ItemCheck(ByVal sender As System.Object, ByVal e As System.Windows.Forms.ItemCheckEventArgs) Handles checkedListBoxHost1.ItemCheck
        Dim coll As CheckedListBox.ObjectCollection = checkedListBoxHost1.Items
        Dim sb As StringBuilder = New StringBuilder()
        For i As Int32 = 0 To coll.Count - 1
            Dim isChecked As Boolean = False
            If i = e.Index Then
                If e.NewValue = CheckState.Checked Then
                    isChecked = True
                Else
                    isChecked = False
                End If
            Else
                isChecked = checkedListBoxHost1.GetItemChecked(i)
            End If
            If isChecked Then
                sb.Append(coll(i)).Append(", ")
            End If
        Next
        Dim res As String
        If sb.Length > 0 Then
            res = sb.ToString(0, sb.Length - 2)
            label1.Text = res
        Else
            res = "<none>"
            label1.Text = "Demo"
        End If
        checkStatusLabel.Text = "Checked: " + res
    End Sub

    Private Sub backColorEditor_SelectedColorChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles backColorEditor.SelectedColorChanged
        label1.BackColor = backColorEditor.SelectedColor
    End Sub

    Private Sub foreColorEditor_SelectedColorChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles foreColorEditor.SelectedColorChanged
        label1.ForeColor = foreColorEditor.SelectedColor
    End Sub

    Private Sub fontEditor_SelectedFontChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles fontEditor.SelectedFontChanged
        If Not _internalFontChange Then
            _internalFontChange = True
            label1.Font = fontEditor.SelectedFont
            numericUpDownHost1.Value = Convert.ToDecimal(fontEditor.SelectedFont.SizeInPoints)
            _internalFontChange = False
        End If
    End Sub

    Private Sub DemoForm_Resize(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Resize
        axWebBrowserHost1.Width = Width - 40
    End Sub

    Private Sub rbLoad_Click(sender As Object, e As EventArgs) Handles rbLoad.Click
        Using dlg As OpenFileDialog = New OpenFileDialog()
            dlg.DefaultExt = ".xml"
            dlg.Filter = "XML files|*.xml|All files|*.*"
            dlg.Title = "Load From XML File"
            If dlg.ShowDialog() = DialogResult.OK Then
                Try
                    C1Ribbon1.Load(dlg.FileName)
                Catch
                    MessageBox.Show("Bad Ribbon XML.", dlg.Title)
                End Try
            End If
        End Using
    End Sub

    Private Sub rbSave_Click(sender As Object, e As EventArgs) Handles rbSave.Click
        Using dlg As SaveFileDialog = New SaveFileDialog()
            dlg.DefaultExt = ".xml"
            dlg.FileName = "ribbon"
            dlg.Filter = "XML files|*.xml|All files|*.*"
            dlg.Title = "Save As XML File"
            If dlg.ShowDialog() = DialogResult.OK Then
                C1Ribbon1.Save(dlg.FileName)
            End If
        End Using
    End Sub

    Private Sub rbClear_Click(sender As Object, e As EventArgs) Handles rbClear.Click
        C1Ribbon1.Tabs.ClearAndDisposeTabs()
    End Sub

    Private Sub C1Ribbon1_ControlHostFromXmlNode(sender As Object, e As ControlHostFromXmlNodeEventArgs) Handles C1Ribbon1.ControlHostFromXmlNode
        Select Case e.NodeName
            Case PictureBoxHost.XmlNodeName
                e.ControlHost = New PictureBoxHost()
            Case NumericUpDownHost.XmlNodeName
                e.ControlHost = New NumericUpDownHost()
            Case FontEditorHost.XmlNodeName
                e.ControlHost = New FontEditorHost()
            Case ColorEditorHost.XmlNodeName
                e.ControlHost = New ColorEditorHost()
            Case CheckedListBoxHost.XmlNodeName
                e.ControlHost = New CheckedListBoxHost()
            Case AxWebBrowserHost.XmlNodeName
                e.ControlHost = New AxWebBrowserHost()
        End Select
    End Sub

    Private Sub C1Ribbon1_ItemDeserialized(sender As Object, e As ItemDeserializedEventArgs) Handles C1Ribbon1.ItemDeserialized
        Select Case e.Item.Name
            Case "exitButton"
                exitButton = CType(e.Item, RibbonButton)
                AddHandler exitButton.Click, AddressOf exitButton_Click
            Case "ribbonApplicationMenu1"
                RibbonApplicationMenu1 = CType(e.Item, RibbonApplicationMenu)
                AddHandler RibbonApplicationMenu1.DoubleClick, AddressOf exitButton_Click
            Case "pictureBoxHost1"
                pictureBoxHost1 = CType(e.Item, PictureBoxHost)
                AddHandler pictureBoxHost1.Click, AddressOf pictureBoxHost1_Click
            Case "numericUpDownHost1"
                numericUpDownHost1 = CType(e.Item, NumericUpDownHost)
                AddHandler numericUpDownHost1.ValueChanged, AddressOf numericUpDownHost1_ValueChanged
            Case "checkedListBoxHost1"
                checkedListBoxHost1 = CType(e.Item, CheckedListBoxHost)
                AddHandler checkedListBoxHost1.ItemCheck, AddressOf checkedListBoxHost1_ItemCheck
            Case "backColorEditor"
                backColorEditor = CType(e.Item, ColorEditorHost)
                AddHandler backColorEditor.SelectedColorChanged, AddressOf backColorEditor_SelectedColorChanged
            Case "foreColorEditor"
                foreColorEditor = CType(e.Item, ColorEditorHost)
                AddHandler foreColorEditor.SelectedColorChanged, AddressOf foreColorEditor_SelectedColorChanged
            Case "fontEditor"
                fontEditor = CType(e.Item, FontEditorHost)
                AddHandler fontEditor.SelectedFontChanged, AddressOf fontEditor_SelectedFontChanged
            Case "axWebBrowserHost1"
                axWebBrowserHost1 = CType(e.Item, AxWebBrowserHost)
                AddHandler exitButton.Click, AddressOf exitButton_Click
                axWebBrowserHost1.Navigate(Application.StartupPath & "\..\..\Html\A plea for C++.htm")
        End Select
    End Sub

    Private Sub C1Ribbon1_RibbonDeserialized(sender As Object, e As EventArgs) Handles C1Ribbon1.RibbonDeserialized
        numericUpDownHost1_ValueChanged(Me, EventArgs.Empty)
        backColorEditor_SelectedColorChanged(Me, EventArgs.Empty)
        foreColorEditor_SelectedColorChanged(Me, EventArgs.Empty)
        fontEditor_SelectedFontChanged(Me, EventArgs.Empty)
        checkedListBoxHost1_ItemCheck(Me, New ItemCheckEventArgs(-1, CheckState.Indeterminate, CheckState.Indeterminate))
    End Sub
End Class

File: ExcludedSamples/NetFramework/Ribbon/CS/ControlHost/DemoForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using C1.Win.C1Ribbon;
using ControlHostLib;

namespace ControlHostDemo
{
    public partial class DemoForm : C1RibbonForm
    {
        private bool _internalFontChange;

        public DemoForm()
        {
            InitializeComponent();

            Font font = label1.Font;
            _internalFontChange = true;
            fontEditor.SelectedFont = font;
            numericUpDownHost1.Value = Convert.ToDecimal(font.SizeInPoints);
            _internalFontChange = false;

            backColorEditor.SelectedColor = label1.BackColor;
            foreColorEditor.SelectedColor = label1.ForeColor;

            axWebBrowserHost1.Navigate(Application.StartupPath +
                @"\..\..\Html\A plea for C++.htm");
        }

        private void exitButton_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void pictureBoxHost1_Click(object sender, EventArgs e)
        {
            MessageBox.Show("You clicked the Earth.", "Congratulations", MessageBoxButtons.OK, MessageBoxIcon.Information);
        }

        private void numericUpDownHost1_ValueChanged(object sender, EventArgs e)
        {
            if (!_internalFontChange)
            {
                Font font = label1.Font;
                font = new Font(font.Name, Convert.ToSingle(numericUpDownHost1.Value), font.Style, font.Unit);
                _internalFontChange = true;
                label1.Font = font;
                fontEditor.SelectedFont = font;
                _internalFontChange = false;
            }
            statusLabel.Text = String.Format("Number is {0}", numericUpDownHost1.Value);
        }

        private void checkedListBoxHost1_ItemCheck(object sender, ItemCheckEventArgs e)
        {
            CheckedListBox.ObjectCollection coll = checkedListBoxHost1.Items;
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < coll.Count; i++)
            {
                bool isChecked = false;
                if (i == e.Index)
                    isChecked = e.NewValue == CheckState.Checked;
                else
                    isChecked = checkedListBoxHost1.GetItemChecked(i);
                if (isChecked)
                {
                    sb.Append(coll[i]).Append(", ");
                }
            }
            string res;
            if (sb.Length > 0)
            {
                res = sb.ToString(0, sb.Length - 2);
                label1.Text = res;
            }
            else
            {
                res = "<none>";
                label1.Text = "Demo";
            }
            checkStatusLabel.Text = "Checked: " + res;
        }

        private void backColorEditor_SelectedColorChanged(object sender, EventArgs e)
        {
            label1.BackColor = backColorEditor.SelectedColor;
        }

        private void foreColorEditor_SelectedColorChanged(object sender, EventArgs e)
        {
            label1.ForeColor = foreColorEditor.SelectedColor;
        }

        private void fontEditor_SelectedFontChanged(object sender, EventArgs e)
        {
            if (!_internalFontChange)
            {
                _internalFontChange = true;
                label1.Font = fontEditor.SelectedFont;
                numericUpDownHost1.Value = Convert.ToDecimal(fontEditor.SelectedFont.SizeInPoints);
                _internalFontChange = false;
            }
        }

        private void DemoForm_Resize(object sender, EventArgs e)
        {
            if (c1Ribbon1.Tabs.Count > 0)
            {
                axWebBrowserHost1.Width = Width - 40;
            }
        }

        private void rbClear_Click(object sender, EventArgs e)
        {
            c1Ribbon1.Tabs.ClearAndDisposeTabs();
        }

        private void rbLoad_Click(object sender, EventArgs e)
        {
            using (OpenFileDialog dlg = new OpenFileDialog())
            {
                dlg.DefaultExt = ".xml";
                dlg.Filter = "XML files|*.xml|All files|*.*";
                dlg.Title = "Load From XML File";
                if (dlg.ShowDialog() == DialogResult.OK)
                {
                    try
                    {
                        c1Ribbon1.Load(dlg.FileName);
                    }
                    catch
                    {
                        MessageBox.Show("Bad Ribbon XML.", dlg.Title);
                    }
                }
            }
        }

        private void rbSave_Click(object sender, EventArgs e)
        {
            using (SaveFileDialog dlg = new SaveFileDialog())
            {
                dlg.DefaultExt = ".xml";
                dlg.FileName = "ribbon";
                dlg.Filter = "XML files|*.xml|All files|*.*";
                dlg.Title = "Save As XML File";
                if (dlg.ShowDialog() == DialogResult.OK)
                {
                    c1Ribbon1.Save(dlg.FileName);
                }
            }
        }

        private void c1Ribbon1_ControlHostFromXmlNode(object sender, ControlHostFromXmlNodeEventArgs e)
        {
            switch (e.NodeName)
            {
                case PictureBoxHost.XmlNodeName:
                    e.ControlHost = new PictureBoxHost();
                    break;
                case NumericUpDownHost.XmlNodeName:
                    e.ControlHost = new NumericUpDownHost();
                    break;
                case FontEditorHost.XmlNodeName:
                    e.ControlHost = new FontEditorHost();
                    break;
                case ColorEditorHost.XmlNodeName:
                    e.ControlHost = new ColorEditorHost();
                    break;
                case CheckedListBoxHost.XmlNodeName:
                    e.ControlHost = new CheckedListBoxHost();
                    break;
                case AxWebBrowserHost.XmlNodeName:
                    e.ControlHost = new AxWebBrowserHost();
                    break;
            }
        }

        private void c1Ribbon1_ItemDeserialized(object sender, ItemDeserializedEventArgs e)
        {
            switch (e.Item.Name)
            {
                case "exitButton":
                    exitButton = (RibbonButton)e.Item;
                    exitButton.Click += exitButton_Click;
                    break;
                case "ribbonApplicationMenu1":
                    ribbonApplicationMenu1 = (RibbonApplicationMenu)e.Item;
                    ribbonApplicationMenu1.DoubleClick += exitButton_Click;
                    break;
                case "pictureBoxHost1":
                    pictureBoxHost1 = (PictureBoxHost)e.Item;
                    pictureBoxHost1.Click += pictureBoxHost1_Click;
                    break;
                case "numericUpDownHost1":
                    numericUpDownHost1 = (NumericUpDownHost)e.Item;
                    numericUpDownHost1.ValueChanged += numericUpDownHost1_ValueChanged;
                    break;
                case "checkedListBoxHost1":
                    checkedListBoxHost1 = (CheckedListBoxHost)e.Item;
                    checkedListBoxHost1.ItemCheck += checkedListBoxHost1_ItemCheck;
                    break;
                case "backColorEditor":
                    backColorEditor = (ColorEditorHost)e.Item;
                    backColorEditor.SelectedColorChanged += backColorEditor_SelectedColorChanged;
                    break;
                case "foreColorEditor":
                    foreColorEditor = (ColorEditorHost)e.Item;
                    foreColorEditor.SelectedColorChanged += foreColorEditor_SelectedColorChanged;
                    break;
                case "fontEditor":
                    fontEditor = (FontEditorHost)e.Item;
                    fontEditor.SelectedFontChanged += fontEditor_SelectedFontChanged;
                    break;
                case "axWebBrowserHost1":
                    axWebBrowserHost1 = (AxWebBrowserHost)e.Item;
                    axWebBrowserHost1.Navigate(Application.StartupPath +
                        @"\..\..\Html\A plea for C++.htm");
                    break;
            }
        }

        private void c1Ribbon1_RibbonDeserialized(object sender, EventArgs e)
        {
            numericUpDownHost1_ValueChanged(this, EventArgs.Empty);
            backColorEditor_SelectedColorChanged(this, EventArgs.Empty);
            foreColorEditor_SelectedColorChanged(this, EventArgs.Empty);
            fontEditor_SelectedFontChanged(this, EventArgs.Empty);

            checkedListBoxHost1_ItemCheck(this, new ItemCheckEventArgs(-1, CheckState.Indeterminate, CheckState.Indeterminate));
        }
    }
}